jwst/group_scale/group_scale.py

```diff
@@ -0,0 +1,56 @@
+from __future__ import division
+
+#
+#  Module for rescaling grouped data for NFRAME
+#  not equal to a power of 2
+#
+
+import logging
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def do_correction(input_model):
+    """
+    Short Summary
+    -------------
+    Rescales all groups in an exposure by FRMDIVSR/NFRAMES, to
+    account for incorrect scaling when on-board frame averaging
+    is done using a value of NFRAMES that is not a power of 2.
+    The divisor for the on-board averaging is always a power of
+    2, so if the number of frames coadded is not a power of 2,
+    the result is incorrect and needs to be rescaled.
+
+    Parameters
+    ----------
+    input_model: data model object
+        science data to be corrected
+
+    Returns
+    -------
+    output_model: data model object
+        rescaled science data
+
+    """
+
+    # Get the meta data values that we need
+    nframes = input_model.meta.exposure.nframes
+    frame_divisor = input_model.meta.exposure.frame_divisor
+    if (nframes is None) or (frame_divisor is None):
```

Comment:
If frame_divisor is None (but nframes is not), one option would be to check whether nframes is a power of two.  If it is, it's likely (or at least possible) that no correction is needed.  If nframes is not a power of two, you could log an error message saying that a correction couldn't be applied but really needs to be.  You could even SWAG it and set frame_divisor to the power of two that is greater than nframes, but of course print a message saying that you guessed at the value of the missing keyword.